project/Component/NefMarkdown/NefMarkdown.swift

```diff
@@ -10,35 +10,77 @@ import Bow
 import BowEffects
 
 public struct Markdown {
-    private let render = Render()
+    public typealias MarkdownEnvironment = RenderMarkdownEnvironment<String>
+    typealias PageOutput  = RenderingOutput<String>.PageOutput
+    typealias PlaygroundOutput  = RenderingOutput<String>.PlaygroundOutput
+    typealias PlaygroundsOutput = RenderingOutput<String>.PlaygroundsOutput
     
     public init() {}
     
-    public func renderPage(content: String) -> EnvIO<RenderMarkdownEnvironment, RenderError, (rendered: String, ast: String)> {
-        render.renderPage(content: content).contramap(\RenderMarkdownEnvironment.renderEnvironment).map { rendered in
-            (rendered: rendered.output, ast: rendered.ast)
-        }^
+    public func page(content: String) -> EnvIO<MarkdownEnvironment, RenderError, (ast: String, rendered: String)> {
```

Comment:
The tuple can be extracted to an alias, as it will probably be used in multiple places. Don't you have already a type to represent this?

Reply:
We have a similar model in an internal layer - `NefCore`. But we should offer a different model in this one :) - I agree to move it to a `typealias`